js/Utility.html

```diff
@@ -12,6 +12,77 @@
   array[index] = array[index].replace(/\s/g, '');
 },
 
+utility.stripItem = (value, index, array) => {
+  // only the quantity and barcode are needed, because the integrity
+  // check will pull the other information from the cache
+  array[index] = {
+    id: array[index].id,
+    description: array[index].description,
+    quantity: array[index].quantity,
+    barcode: array[index].barcode,
+  };
+},
+
+utility.itemIntegrityCheck = (item, index, array) => {
```

Comment:
This is too specific for the utility file.  This is a very specific set of instructions.  I believe it belongs to `js/app/pages/Form`.

Reply:
Reviewing this again, I think you're writing a `Item` constructor here, which is good, but let's just go ahead and write that.  The word "check" in `itemIntegrityCheck` implies this function returns a `bool`.  But as you state later, you are actually rebuilding an item from a lookup key.  That pattern should be `const someItem = Item(key);` where `Item` is an item constructor that knows how to make a valid item.